### DISCUSSION TITLE
Filterset issue - unrecognized field type FileField

### DISCUSSION THREAD

**Discussion by andychoi** [carltongibson/django-filter#1506] [General]
Here is error.

> AssertionError: ProjectFilterSet resolved field 'attachment' with 'exact' lookup to an unrecognized field type FileField. Try adding an override to 'Meta.filter_overrides'. See: https://django-filter.readthedocs.io/en/main/ref/filterset.html#customise-filter-generation-with-filter-overrides

My class has FileField, but I didn't include it in filter class as below. I tried to exclude, but still not working

```
class ProjectFilter(django_filters.FilterSet):

    class Meta:
        model = Project
        exclude = ['attachment']
        fields = {
            'title'         : ['icontains'],
            'description'   : ['icontains'],
        }

```

**Comment by carltongibson** [comment]
You can't provide `fields` and `exclude` at the same time. The `fields` setting should be sufficient? 🤔

**Comment by DavidHostler** [comment]
I was having this issue for awhile too.

The issue is that certain fields (namely ImageField and FileField) are not recognized as valid model fields by this package. 
I solved it on my system by hand-coding the fields in the django-filter package itself. It turns out that it worked for me, and the rest is history.

If the author of this repo is alright with it, I would like to submit a pull request and see if we can integrate it into the package.

**Reply by carltongibson** [reply]
See also  #1478
